### PR TITLE
Fix `ut_lind_fs_mkdir_invalid_modebits`

### PR DESCRIPTION
**Pull request by ChinmayShringi**
Fixes # (issue)

This update modifies `mkdir_syscall` to properly handle and reject invalid mode bits when creating a directory. If the mode includes bits outside the valid range (`0o0000` to `0o0777`), the syscall now returns an `EPERM` error, indicating that the permission bits are not valid. This change ensures consistent behavior in accordance with POSIX standards, preventing directories from being created with unsupported or incorrect permissions.

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- `cargo test ut_lind_fs_mkdir_invalid_modebits`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, lind-project)

### CODE REVIEW

File: src/safeposix/syscalls/fs_calls.rs [hunk 1]
```diff
@@ -71,6 +71,9 @@ impl Cage {
     *   mkdir() will return 0 when success and -1 when fail 
     */
     pub fn mkdir_syscall(&self, path: &str, mode: u32) -> i32 {
+        if (mode & !0o777) != 0 {
+            return syscall_error(Errno::EPERM, "mkdir", "Invalid mode bits specified");
+        }
```
Comment (Yaxuan-w) [inline src/safeposix/syscalls/fs_calls.rs]: what is currently rawposix returned in this situation?

Comment (ChinmayShringi) [reply src/safeposix/syscalls/fs_calls.rs]: Currently it return a -1 value

Comment (Yaxuan-w) [reply src/safeposix/syscalls/fs_calls.rs]: Could you test with [`get_errno()`](https://github.com/Lind-Project/RawPOSIX/blob/98fedeab93440d63b69e044ca81a3da5f84220d5/src/interface/misc.rs#L139) to see the errno returned by libc::mmap?

Sth like:
```rust
let ret = libc::mmap(...)
if ret < 0 {
  println!(get_errno());
}
```

**Review by Yaxuan-w — APPROVED** [review]
LGTM

File: src/safeposix/syscalls/fs_calls.rs (hunk 1 above)
Comment (Yaxuan-w) [reply src/safeposix/syscalls/fs_calls.rs]: This case should be handled in rust libc as well... So we don't need those checks

**Review by Yaxuan-w — CHANGES_REQUESTED** [review]
sry for the confusion

**Review by Yaxuan-w — APPROVED** [review]
lgtm

**Pull request merged**